src/HazelcastClient.ts

```diff
@@ -146,7 +146,12 @@ export class HazelcastClient {
         this.loggingService = new LoggingService(this.config.customLogger,
             this.config.properties['hazelcast.logging.level'] as string);
         const logger = this.loggingService.getLogger();
-        this.schemaService = new SchemaService(() => this.invocationService, logger);
+        this.schemaService = new SchemaService(
```

Comment:
You can move clusterService and clusterFailoverService construction above and use clusterService directly: 

```diff
diff --git a/src/HazelcastClient.ts b/src/HazelcastClient.ts
index a39d6eb5..9d35a447 100644
--- a/src/HazelcastClient.ts
+++ b/src/HazelcastClient.ts
@@ -146,10 +146,16 @@ export class HazelcastClient {
         this.loggingService = new LoggingService(this.config.customLogger,
             this.config.properties['hazelcast.logging.level'] as string);
         const logger = this.loggingService.getLogger();
+        this.clusterFailoverService = this.initClusterFailoverService();
+        this.clusterService = new ClusterService(
+            this.config,
+            logger,
+            this.clusterFailoverService
+        );
         this.schemaService = new SchemaService(
-            this.config, 
-            () => this.clusterService, 
-            () => this.invocationService, 
+            this.config,
+            this.clusterService,
+            () => this.invocationService,
             logger
         );
         this.serializationService = new SerializationServiceV1(this.config.serialization, this.schemaService);
@@ -166,12 +172,6 @@ export class HazelcastClient {
             this.config.lifecycleListeners,
             logger
         );
-        this.clusterFailoverService = this.initClusterFailoverService();
-        this.clusterService = new ClusterService(
-            this.config,
-            logger,
-            this.clusterFailoverService
-        );
         this.connectionRegistry = new ConnectionRegistryImpl(
             this.config.connectionStrategy.asyncStart,
             this.config.connectionStrategy.reconnectMode,

```

Reply:
Fixed